physicsnemo/experimental/models/dit/dit.py

```diff
@@ -0,0 +1,453 @@
+        x = x + gate_mlp.unsqueeze(1) * self.mlp(modulate(self.norm2(x), shift_mlp, scale_mlp))
+        return x
+
+
+class FinalLayer(nn.Module):
+    """
+    The final layer of DiT.
+    """
+    def __init__(self, hidden_size, patch_size, out_channels):
+        super().__init__()
+        self.norm_final = nn.LayerNorm(hidden_size, elementwise_affine=False, eps=1e-6)
+        self.linear = nn.Linear(hidden_size, patch_size * patch_size * out_channels, bias=True)
+        self.adaLN_modulation = nn.Sequential(
+            nn.SiLU(),
+            nn.Linear(hidden_size, 2 * hidden_size, bias=True)
+        )
+
+    def forward(self, x, c):
+        shift, scale = self.adaLN_modulation(c).chunk(2, dim=1)
+        x = modulate(self.norm_final(x), shift, scale)
+        x = self.linear(x)
+        return x
+
+
+class DiT(Module):
+    """
+    Diffusion model with a Transformer backbone.
+    """
+    def __init__(
+        self,
+        input_size: Union[int, Tuple[int, int]] = (32,32),
+        patch_size: int = 2,
+        in_channels: int = 4,
+        out_channels: Union[None, int] = None,
+        hidden_size: int = 1152,
+        depth: int = 28,
+        num_heads: int = 16,
+        mlp_ratio: float = 4.0,
+        use_transformer_engine_attn: bool = False,
+        class_dropout_prob=0.1,
+        condition_dim=None,
+        learn_sigma=True,
+        embedding_type='sin-cos',
+    ):
+        """
+        Initializes the DiT model.
+
+        Args:
+            input_size (Tuple[int, int]): Height and width of the input images. Default is (32, 32).
+            patch_size (int): The size of each image patch. Defaults to 2.
+            in_channels (int): The number of input channels. Defaults to 4.
+            out_channels (None, int): The number of output channels. Defaults to None. If None, it is either in_channels or 2*in_channels based on learn_sigma
+            hidden_size (int): The dimensionality of the transformer embeddings. Defaults to 1152.
+            depth (int): The number of transformer blocks. Defaults to 28.
+            num_heads (int): The number of attention heads. Defaults to 16.
+            mlp_ratio (float): The ratio of the MLP hidden dimension to the embedding dimension. Defaults to 4.0.
+            use_transformer_engine_attn (bool): If True, use MultiheadAttention from transformer_engine. Defaults to False.
+            class_dropout_prob (float): The dropout probability for classifier-free guidance. Defaults to 0.1.
+            condition_dim (int, optional): The number of classes for conditioning. If None, the model is unconditional. Defaults to None.
+            learn_sigma (bool): If True, the model learns the variance of the diffusion process. Defaults to True.
+            embedding_type (str): The type of positional embedding to use. 'sin-cos' for fixed sinusoidal embeddings,
+                                  'positional' for learnable positional embeddings. Defaults to 'sin-cos'.
```

Comment:
I think the learnable pos embed should be called "learnable". rather than "positional"